underscore.js

```diff
@@ -1239,6 +1246,12 @@
   // Shortcut function for checking if an object has a given property directly
   // on itself (in other words, not on a prototype).
   _.has = function(obj, key) {
+    if (_.isString(key) && key.indexOf('.') !== -1) {
+      var keys = key.split('.');
+      key = keys.pop();
+      var path = keys.join('.');
+      obj = resolveObject(obj, path);
+    }
```

Comment:
I guess this would mean that you could never check for the existence of a key which contains a dot:

```js
_.has({ 'my.key': true}, 'my.key') === false
```

You could make it more like:

```js
return obj != null && (hasOwnProperty.call(obj, key) || resolveObject(obj, path));
```

*(There'd be an issue there since resolve object returns the value, not the bool showing existence of the key, but you get the idea.)*